### ISSUE TITLE
jsdoc/require-template doesn't work with a default value for a type parameter

### ISSUE THREAD

**Issue by alexander-akait** [gajus/eslint-plugin-jsdoc#1418]
## Expected behavior

No error

## Actual behavior

Error

```
X:X  error  Missing @template T  jsdoc/require-template
```

If change `@template {string | Buffer} [T=string]` to `@template {string | Buffer} T` all works fine.

## ESLint Config

```js
{
  "jsdoc/require-template": "error",
}
```

## ESLint sample

```js
/**
 * @template {string | Buffer} [T=string]
 * @typedef {object} Dirent
 * @property {() => boolean} isFile true when is file, otherwise false
 * @property {() => boolean} isDirectory true when is directory, otherwise false
 * @property {() => boolean} isBlockDevice true when is block device, otherwise false
 * @property {() => boolean} isCharacterDevice true when is character device, otherwise false
 * @property {() => boolean} isSymbolicLink true when is symbolic link, otherwise false
 * @property {() => boolean} isFIFO true when is FIFO, otherwise false
 * @property {() => boolean} isSocket true when is socket, otherwise false
 * @property {T} name name
 * @property {string} parentPath path
 * @property {string=} path path
 */
```

## Environment

- Node version: v22.13.1
- ESLint version v9.28.0
- `eslint-plugin-jsdoc` version: v51.2.2

**Comment by alexander-akait** [comment]
@brettz9 Thanks for fast fix ⭐
